core/libp2p/network/impl/network_impl.hpp

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_NETWORK_IMPL_HPP
+#define KAGOME_NETWORK_IMPL_HPP
+
+#include "libp2p/network/connection_manager.hpp"
+#include "libp2p/network/network.hpp"
+
+namespace libp2p::network {
+
+  class NetworkImpl : public Network {
+   public:
+    ~NetworkImpl() override = default;
+
+    NetworkImpl(std::unique_ptr<ListenerManager> listener,
+                std::unique_ptr<Dialer> dialer,
+                std::shared_ptr<ConnectionManager> cmgr);
+
+    // Closes all connections to a given peer
+    void closeConnections(const peer::PeerId &p) override;
+
+    // Getter for Dialer associated with this Network
+    Dialer &getDialer() override;
+
+    // Getter for Listener associated with this Network
```

Comment:
```suggestion
    /// Getter for Listener associated with this Network
```
And others in this file

Reply:
Why haven't you? This is format for docs, as I know